daemon/info_unix.go

```diff
@@ -184,6 +185,51 @@ func (daemon *Daemon) fillPlatformVersion(ctx context.Context, v *types.Version,
 		}
 		log.G(ctx).WithError(err).Warn("Failed to fill rootless version")
 	}
+
+	if err := daemon.fillModuleversion(ctx, v); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (daemon *Daemon) fillModuleversion(ctx context.Context, v *types.Version) error {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		log.G(ctx).Warn("Failed to read build info")
+		return nil
+	}
+
+	if len(v.Components) > 0 && v.Components[0].Name == "Engine" && v.Components[0].Details != nil {
+		v.Components[0].Details["Module"] = info.Main.Path
+		v.Components[0].Details["ModuleVersion"] = info.Main.Version
+	}
+
+	for _, dep := range info.Deps {
+		var name string
+		switch dep.Path {
+		case "github.com/moby/moby/v2":
+			name = "moby"
+		case "github.com/moby/moby/api":
+			name = "moby/api"
```

Comment:
Adding the version of the `api` module will be confusing; I don't think it adds anything, because the version of the module does not determine the API version that's supported by the daemon. I can update the API module to (e.g.) get bug-fixes in, but it won't make the daemon support new features of the API. The version of the API supported by the daemon is (currently) controlled by the `MinAPIVersion` and `MaxAPIVersion` consts. https://github.com/moby/moby/blob/f835ff6987c5f907eebbc4d709fd6400e41cfe8b/daemon/config/config.go#L58-L66